### PR TITLE
Improve saltssh error parsing

### PR DESCRIPTION
**Pull request by mcalmer**
## What does this PR change?

salt ssh sometimes put usefull messages in stdout while stderr
is an empty string. But Options.of("") is still "present" which
result in showing an empty sting as error message

## GUI diff

No difference.

Before:

image

After:

Screenshot_20200610_120127

- [x] **DONE**

## Documentation
- No documentation needed: **internal**

- [x] **DONE**

## Test coverage
- No tests: **one out of a million parsing problems**

- [x] **DONE**

## Links

Fixes https://github.com/SUSE/spacewalk/issues/11557
Tracks #

- [ ] **DONE**

## Changelogs

If you don't need a changelog check, please mark this checkbox:

- [ ] No changelog needed

If you uncheck the checkbox after the PR is created, you will need to re-run `changelog_test` (see below)

## Re-run a test

If you need to re-run a test, please mark the related checkbox, it will be unchecked automatically once it has re-run:

- [ ] Re-run test "changelog_test"
- [ ] Re-run test "backend_unittests_pgsql"
- [ ] Re-run test "java_lint_checkstyle"
- [ ] Re-run test "java_pgsql_tests" 
- [ ] Re-run test "ruby_rubocop"
- [ ] Re-run test "schema_migration_test_oracle"
- [ ] Re-run test "schema_migration_test_pgsql"
- [ ] Re-run test "susemanager_unittests"
- [ ] Re-run test "javascript_lint"
- [ ] Re-run test "spacecmd_unittests"

### CODE REVIEW

**Comment by mcalmer** [comment]
@ncounter I wonder if we should also change 

```
    public static String decodeSaltErr(SaltError saltErr) {
        Optional<String> errorMessage = SaltUtils.decodeStdMessage(saltErr, "stderr");
        Optional<String> outMessage = !errorMessage.isPresent() ?
                SaltUtils.decodeStdMessage(saltErr, "stdout") : errorMessage;
        Optional<String> returnMessage = !outMessage.isPresent() ?
                SaltUtils.decodeStdMessage(saltErr, "return") : outMessage;
        return returnMessage.orElseGet(() -> saltErr.toString());
    }
```

I think we could also have the case where "stderr" contains something (not important) while "stdout" has real good info.

Should we just concat stderr and stdout to get all the info?

**Comment by ncounter** [comment]
> 
> I think we could also have the case where "stderr" contains something (not important) while "stdout" has real good info.
> 
> Should we just concat stderr and stdout to get all the info?


I see. Mmm, I am having hard times to find a proper documentation of returned values/messages in these cases actually, not sure if you have any better chance. Anyway, generally speaking, I'd say `stdout` is supposed to contain `info` messages, and `stderr` only `error` messages. If they are exclusively present, it is fine to always present the one which has value. But what to do if they are both present? `info` message (blue bubble) or `error` message (red bubble)? I'd say that if the `stderr` is populated it means there was an error, so either we show first the `stderr` message, and we add for more info the `stdout` message, or we keep them separated like two completely separated bubble, a blue one for `stdout` and a red one for `stderr`.

Bottom line: yes, if we have some feedback from the return of `salt-ssh`, better to show all than some or none. :+1:

**Comment by mcalmer** [comment]
Screenshot_20200610_105331

@ncounter Please guess what of this message come from STDERR and what from STDOUT ?

**Comment by ncounter** [comment]
> @ncounter Please guess what of this message come from STDERR and what from STDOUT ?

LOL!

I bet: before `Error:` it is from `stdout`, after is from `stderr`.

That's totally an unclear message. Let's keep it as an error the entire message. Or, one last option, if we have a mix of the two, we can use a `warning` message :stuck_out_tongue_closed_eyes:

**Comment by mcalmer** [comment]
> > @ncounter Please guess what of this message come from STDERR and what from STDOUT ?
> 
> LOL!
> 
> I bet: before `Error:` it is from `stdout`, after is from `stderr`.

Sorry, wrong!

Before `Error:` ("Warning: Permanently added 'suma-refhead-min-centos7.mgr.suse.de,2620:113:80c0:8300:a8b2:93ff:fe00:34' (ECDSA) to the list of known hosts.") comes as STDERR.
STDOUT was: "ERROR: No scp binary. openssh-clients package required"

This is the reason why I want to show both.

**Comment by ncounter** [comment]
:man_facepalming:

Ok, agreed! If `stderr` is present, everything is an error.
